### PR TITLE
Fix data marker update

### PR DESCRIPTION
**Pull request by smbea**
Related to https://github.com/camunda/camunda-modeler/issues/3815

As per https://github.com/camunda/camunda-modeler/issues/3815#issuecomment-1733038161

### CODE REVIEW

File: test/spec/components/Group.spec.js [hunk 1]
```diff
@@ -43,6 +43,11 @@ describe('<Group>', function() {
     container = TestContainer.get(this);
   });
 
+  beforeEach(() => {
+    const requestAnimationFrameSpy = sinon.fake.returns(cb => cb());
+    global.window.requestAnimationFrame = requestAnimationFrameSpy();
+  });
```
Comment (barmac) [inline test/spec/components/Group.spec.js:49]: The spy is not cleaned up properly, we should use sth like:

```javascript
afterEach(() => global.window.requestAnimationFrame.restore());
```

File: src/components/Group.js [hunk 2]
```diff
@@ -58,25 +58,29 @@ export default function Group(props) {
 
   // set edited state depending on all entries
   useEffect(() => {
-    const hasOneEditedEntry = entries.find(entry => {
-      const {
-        id,
-        isEdited
-      } = entry;
+    const scheduled = requestAnimationFrame(() => {
```
Comment (barmac) [inline src/components/Group.js:61]:
```suggestion
    // TODO(@barmac): replace with CSS when `:has()` is supported in all major browsers, or rewrite as in https://github.com/camunda/camunda-modeler/issues/3815#issuecomment-1733038161
    const scheduled = requestAnimationFrame(() => {
```

File: test/spec/components/Group.spec.js [hunk 3]
```diff
@@ -43,6 +43,11 @@ describe('<Group>', function() {
...
+
```
Comment (barmac) [inline test/spec/components/Group.spec.js:50]:
```suggestion

  afterEach(() => global.window.requestAnimationFrame.restore());
```

**Comment by barmac** [comment]
I applied my suggestions from code review; after the CI finishes let's merge this!

**Review by barmac — APPROVED** [review]

File: test/spec/components/Group.spec.js (hunk 1 above)
Comment (smbea) [reply test/spec/components/Group.spec.js:49]: Seems like this doesn't work: https://github.com/bpmn-io/properties-panel/actions/runs/6300214448/job/17102537862?pr=293#step:6:243

Comment (barmac) [reply test/spec/components/Group.spec.js]: Indeed, it looks like fakes don't have `restore` even though it's in the types file.

https://sinonjs.org/releases/latest/fakes/

**Comment by barmac** [comment]
Ok now the tests with `restore` work fine locally, and should also pass on CI.

**Pull request merged**